### PR TITLE
WI #1997 Always create code element for ACCEPT statement

### LINKED ISSUE #1997: Allow empty ACCEPT statement in grammar
Child of #1835.

Modify the ANTLR grammar to allow `ACCEPT` as a stand alone keyword. The code element need to be created and then checked manually.

### PR DESCRIPTION
**Pull request by fm-117**
Fixes #1997.

### CODE REVIEW

File: TypeCobol/AntlrGrammar/CobolCodeElements.g4 [hunk 1]
```diff
@@ -3843,13 +3843,13 @@ sentenceEnd:
...
 
-acceptDataTransfer:
-	ACCEPT alphanumericStorageArea (FROM mnemonicForEnvironmentNameReferenceOrEnvironmentName)?;
+fromEnvironment:
+	FROM mnemonicForEnvironmentNameReferenceOrEnvironmentName;
 
-acceptSystemDateTime:
-	ACCEPT alphanumericStorageArea FROM ((DATE yyyyMmDd?) | (DAY yyyyDdd?) | DAY_OF_WEEK | TIME);
+fromSystemDateTime:
+	FROM ((DATE yyyyMmDd?) | (DAY yyyyDdd?) | DAY_OF_WEEK | TIME);
```
Comment (delevoye) [inline TypeCobol/AntlrGrammar/CobolCodeElements.g4:3852]: Is it possible to have the message coming from ANTLR?
As you said: maybe by keeping the initial grammar but changing the builder only.

File: TypeCobol/Compiler/Parser/CodeElementBuilder/CobolStatementsBuilder.cs [hunk 2]
```diff
@@ -52,20 +52,19 @@ public CobolStatementsBuilder(CobolWordsBuilder cobolWordsBuilder, CobolExpressi
 		 // ACCEPT STATEMENT //
 		//////////////////////
 
-		internal AcceptFromInputDeviceStatement CreateAcceptDataTransferStatement(CodeElementsParser.AcceptDataTransferContext context) {
+		internal AcceptFromInputDeviceStatement CreateAcceptDataTransferStatement(ReceivingStorageArea receivingStorageArea, CodeElementsParser.FromEnvironmentContext context) {
```
Comment (delevoye) [inline TypeCobol/Compiler/Parser/CodeElementBuilder/CobolStatementsBuilder.cs:55]: Is the name of the function still senseful?

**Review by delevoye — CHANGES_REQUESTED** [review]

File: TypeCobol/AntlrGrammar/CobolCodeElements.g4 (hunk 1 above)
Comment (fm-117) [reply TypeCobol/AntlrGrammar/CobolCodeElements.g4]: Using the original grammar with a modified builder does not seem to work. We end up with unwanted calls to `EnterAcceptStatement`, themselves resulting into `AcceptStatement` code elements with no consumed tokens. Although I don't have the explanation for those extra calls, it seems that the modified grammar is more reliable, producing the correct number of code elements.

I've changed it using your suggestion on the syntax for the optional fragement.

File: TypeCobol/Compiler/Parser/CodeElementBuilder/CobolStatementsBuilder.cs (hunk 2 above)
Comment (fm-117) [reply TypeCobol/Compiler/Parser/CodeElementBuilder/CobolStatementsBuilder.cs:55]: As it still produces the instance of the code element, i's say yes 😄.

File: TypeCobol/AntlrGrammar/CobolCodeElements.g4 (hunk 1 above)
Comment (delevoye) [reply TypeCobol/AntlrGrammar/CobolCodeElements.g4]: If the last solution works, it should be fine as it stays understandable.

**Review by delevoye — APPROVED** [review]

**Review by mayanje — APPROVED** [review]

**Review by rooksdo — APPROVED** [review]

**Pull request merged**